### ISSUE TITLE
Enqueue request from script instead of command

### ISSUE THREAD

**Issue by Nirei** [savonet/liquidsoap#949]
I'm trying to write my very first liquidsoap program. It goes something like this:

```
sounds_path = "../var/sounds"

# Log file
set("log.file.path","var/log/liquidsoap.log")
set("harbor.bind_addr", "127.0.0.1")
set("harbor.timeout", 5)
set("harbor.verbose", true)
set("harbor.reverse_dns", false)

silence = blank()
queue = request.queue()

def play(~protocol, ~data, ~headers, uri) =
    request.push("#{sounds_path}#{uri}")
    http_response(protocol=protocol, code=20000)
end

harbor.http.register(port=8080, method="POST", "^/(?!\0)+", play)

stream = fallback(track_sensitive=false, [queue, silence])
...output.whatever...
```

And I was wondering if there is any way to push to the queue from the harbor callback.

Else, how should I proceed about making requests originate from HTTP calls? I really want to avoid telnet. My final objective is having an endpoint that I can call to make my stream play a file on demand and be silent the rest of the time.

**Comment by smimram** [comment]
Actually, this is quite a good question and I am not sure this is currently possible without telnet (whereas it really should and telnet should be registered from Liquidsoap...)

@toots ?

**Comment by Nirei** [comment]
So I was thinking I could try to make myself useful.

After an interesting evening getting remotely aquainted with the codebase I tried putting this in the initializer of the ``Req_queue.queue`` class:

```ocaml
      Lang.add_builtin "request.queue.push"
        ~category:(Lang.string_of_category Lang.Liquidsoap)
        ~descr:"Add a request to a queue."
        [ "", Lang.request_t k, None, None ]
        Lang.unit_t
        (fun p ->
            let r = Lang.to_request (Lang.assoc "" 2 p) in
            self#push_request r;
            fun (_:Lang.t) -> Lang.unit)
```

But no use: my builtin is not recognized when I try to call it from my liquidsoap script.

Also, I didn't quite grasp why my ``fun p -> ...`` kept complaining about its return type having to be a function instead of pure ``Lang.unit``.

If you are interested and can give me some pointers I can try to write a PR for this?

Very cool software anyway, folks, and thanks for your time. : )

**Comment by smimram** [comment]
#1010 should be quite helpful for this one, hold on.

**Comment by smimram** [comment]
This should be solved in #1013 which I guess is going to be merged soon (feel free to try before that...)

**Comment by Nirei** [comment]
Sorry, didn't find out about slack invite until very recently. :(
